File: Napakalaki/src/napakalaki/Dice.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;

public class Dice{
    private static Dice instance = null;

    private Dice(){

    }

    public static Dice getInstance() {
        if(instance == null) {
          instance = new Dice();
        }
        return instance;
    }

    public int nextNumber(){
        int x = (int) (Math.random() * 6); 
        return x;
    }
}

File: NapakalakiRuby/lib/dice.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  require "singleton"
	class Dice
		include Singleton
    

		def nextNumber
			r = Random.rand(1...6)
			return r
		end
    
    
  end
  
end
  

